opengrok-indexer/src/main/java/org/opengrok/indexer/history/GitRepository.java

```diff
@@ -95,6 +96,28 @@
      */
     private static final Version MINIMUM_VERSION = new Version(2, 1, 2);
 
+    /**
+     * This is a static replacement for 'working' field. Effectively, check if git is working once in a JVM
+     * instead of calling it for every GitRepository instance.
+     */
+    private static final LazilyInstantiate<Boolean> GIT_IS_WORKING = LazilyInstantiate.using(
+            () -> isGitWorking(getCommand(GitRepository.class.getCanonicalName(), CMD_PROPERTY_KEY, CMD_FALLBACK)));
+
+    private static boolean isGitWorking(String repoCommand) {
```

Comment:
As this is not used anywhere else it does not need to be generic (wrt to the repository command). You can get the command inside this function and do just 

```
LazilyInstantiate.using(GitRepository::isGitWorking);
```